weave

```diff
@@ -910,21 +923,26 @@ ipam_cidrs() {
             else
                 IPAM_URL=/ip/$CONTAINER_ID/"${arg#net:}"
             fi
-            CIDR=$(call_weave POST $IPAM_URL$CHECK_ALIVE) || return 1
+            CIDR=$(call_weave $METHOD $IPAM_URL$CHECK_ALIVE) || return 1
             if [ "$CIDR" = "404 page not found" ] ; then
-                echo "IP address allocation must be enabled to use 'net:'" >&2
-                return 1
+                if [ "$METHOD" = "POST" ] ; then
+                    echo "IP address allocation must be enabled to use 'net:'" >&2
+                    return 1
+                fi
             elif ! is_cidr "$CIDR" ; then
                 echo "$CIDR" >&2
                 return 1
+            else
+                IPAM_CIDRS="$IPAM_CIDRS $CIDR"
+                ALL_CIDRS="$ALL_CIDRS $CIDR"
             fi
-            IPAM_CIDRS="$IPAM_CIDRS $CIDR"
-            ALL_CIDRS="$ALL_CIDRS $CIDR"
         else
-            # This is a plain IP address; warn if it clashes but carry on
-            command_exists netcheck && netcheck --ignore-iface=$BRIDGE $arg || true
-            if container_ip $CONTAINER_NAME 2>/dev/null ; then
-                http_call $CONTAINER_IP:$HTTP_PORT PUT /ip/$CONTAINER_ID/${arg%/*} >&2
+            if [ "$METHOD" = "POST" ] ; then
+                # Allocation of a plain IP address; warn if it clashes but carry on
```

Comment:
This is not an _allocation_; the user has passed in a specific IP address.    "assignment" ?